azurerm/internal/services/compute/windows_virtual_machine_resource.go

```diff
@@ -0,0 +1,1004 @@
+	} else if priority == compute.Spot {
+		return fmt.Errorf("An `eviction_policy` must be specified when `priority` is set to `Spot`")
+	}
+
+	if v, ok := d.GetOk("license_type"); ok {
+		params.LicenseType = utils.String(v.(string))
+	}
+
+	if v, ok := d.Get("max_bid_price").(float64); ok && v > 0 {
+		if priority != compute.Spot {
+			return fmt.Errorf("`max_bid_price` can only be configured when `priority` is set to `Spot`")
+		}
+
+		params.BillingProfile = &compute.BillingProfile{
+			MaxPrice: utils.Float(v),
+		}
+	}
+
+	if v, ok := d.GetOk("proximity_placement_group_id"); ok {
+		params.ProximityPlacementGroup = &compute.SubResource{
+			ID: utils.String(v.(string)),
+		}
+	}
+
+	if v, ok := d.GetOk("timezone"); ok {
+		params.VirtualMachineProperties.OsProfile.WindowsConfiguration.TimeZone = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("zone"); ok {
+		params.Zones = &[]string{
+			v.(string),
+		}
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, params)
+	if err != nil {
+		return fmt.Errorf("Error creating Windows Virtual Machine %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Windows Virtual Machine %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Windows Virtual Machine %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Error retrieving Windows Virtual Machine %q (Resource Group %q): `id` was nil", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+	return resourceWindowsVirtualMachineRead(d, meta)
+}
+
+func resourceWindowsVirtualMachineRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.VMClient
+	disksClient := meta.(*clients.Client).Compute.DisksClient
+	networkInterfacesClient := meta.(*clients.Client).Network.InterfacesClient
+	publicIPAddressesClient := meta.(*clients.Client).Network.PublicIPsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := ParseVirtualMachineID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Windows Virtual Machine %q was not found in Resource Group %q - removing from state!", id.Name, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Windows Virtual Machine %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*resp.Location))
```

Comment:
I think this could be
```suggestion
		d.Set("location", azure.NormalizeLocation(*location))
```